M2/Macaulay2/d/CMakeLists.txt

```diff
@@ -144,8 +144,9 @@ add_library(M2-interpreter OBJECT ${CLIST} ${CXXLIST} ${TAGS}
   $<$<BOOL:${WITH_XML}>:xml-c.c xml-c.h>
   $<$<BOOL:${WITH_PYTHON}>:python-c.c>)
 
+target_include_directories(M2-interpreter PUBLIC ${FFI_INCLUDE_DIRS})
```

Comment:
Why should this be public?

Reply:
I had trouble understanding cmake docs on that, which say

The INTERFACE, PUBLIC and PRIVATE keywords are required to specify the [scope](https://cmake.org/cmake/help/latest/manual/cmake-buildsystem.7.html#target-usage-requirements) of the following arguments. 
PRIVATE and PUBLIC items will populate the [INCLUDE_DIRECTORIES](https://cmake.org/cmake/help/latest/prop_tgt/INCLUDE_DIRECTORIES.html#prop_tgt:INCLUDE_DIRECTORIES) property of `target`. PUBLIC and INTERFACE items will populate the [INTERFACE_INCLUDE_DIRECTORIES](https://cmake.org/cmake/help/latest/prop_tgt/INTERFACE_INCLUDE_DIRECTORIES.html#prop_tgt:INTERFACE_INCLUDE_DIRECTORIES) property of `target`. 

Should it rather be PRIVATE?

Reply:
done

Reply:
`PUBLIC` means if another component depends on the current component (`M2-interpreter` in this case), they will also have these compiler flags passed onto them, `PRIVATE` means they don't.